README.md

```diff
@@ -32,3 +32,47 @@ or run only tests using
 ```bash
 vendor/bin/phing tests
 ```
+
+### Running development version
+
+The executable is `bin/phpstan`. Rut it as
+```bash
+bin/phpstan analyze -c your-config.neon -l 8 path-to-analyze
+```
+As you change the source code it immediately takes effect, no re-building is required.
+
+
+### Debugging
+
+1. Make sure XDebug is installed and configured using any HOWTO available.
+2. Add `--xdebug` option when running phpstan. Without it PHPStan turns the debugger off at runtime.
+
+Now you are ready to set breakpoints on rules and have fun.
+
+
+### Fixing code style
+
+To detect code style issues, run:
+```bash
+vendor/bin/phing cs
+```
+
+This requires PHP 7.4. Otherwise the check will be skipped and succeed silently.
+
+And then to fix code style, run:
+
+```bash
+vendor/bin/phing cs-fix
+```
+This does not depend on PHP version.
+
+
+### Running tests
+
+If you have XDebug enabled and run PHPUnit directly instead of `phing`, make sure to set `PHPSTAN_ALLOW_XDEBUG` environment variable to 1 before running PHPUnit. Otherwise PHPStan will detect XDebug and try to disable it. It would restart `phpunit` with arguments of `phpstan` which will break.
+
+Run:
+```bash
+export PHPSTAN_ALLOW_XDEBUG=1
+vendor/bin/phpunit --bootstrap=tests/bootstrap.php --stop-on-failure tests
```

Comment:
does `PHPSTAN_ALLOW_XDEBUG=1 vendor/bin/phpunit --bootstrap=tests/bootstrap.php --stop-on-failure tests` work, or it needs to be exported?

Reply:
This should work as well, but I find `export` easier to read.

Reply:
Not sure what you're referring to, this variable isn't anywhere in the codebase: https://github.com/phpstan/phpstan-src/search?q=PHPSTAN_ALLOW_XDEBUG&unscoped_q=PHPSTAN_ALLOW_XDEBUG

Reply:
but `export` sets variable for whole terminal session, if i prepend it to command, it should be set only for that process

Reply:
Run this with xdebug enabled:
```bash
vendor/bin/phpunit --bootstrap=tests/bootstrap.php --stop-on-failure --filter=testConfigurationAutoDiscovery tests/PHPStan/Command/AnalyseCommandTest.php
```

The following will happen:

1. `--xdebug` option is not passed to `phpstan` by the test.
2. CommandHelper checks `!$allowXdebug` and calls XdebugHandler::check().
3. XdebugHandler checks getenv($this->envAllowXdebug) and finds that it is not set.
4. XdebugHandler then restarts the command using the only command it can find:
```bash
'/usr/bin/php7.3' '-n' '-c' '/tmp/rIqFYd' '/home/alexey/work/phpstan-src/vendor/phpunit/phpunit/phpunit' '--bootstrap=tests/bootstrap.php' '--stop-on-failure' '--filter=testConfigurationAutoDiscovery' 'tests/PHPStan/Command/AnalyseCommandTest.php' '--ansi'
```
5. PHPUnit dies as it cannot recognize the `--ansi` option.

The stack of this restart is
```
/home/alexey/work/phpstan-src/vendor/composer/xdebug-handler/src/XdebugHandler.php:136
/home/alexey/work/phpstan-src/src/Command/CommandHelper.php:51
/home/alexey/work/phpstan-src/src/Command/AnalyseCommand.php:126
/home/alexey/work/phpstan-src/vendor/symfony/console/Command/Command.php:255
/home/alexey/work/phpstan-src/vendor/symfony/console/Tester/CommandTester.php:76
/home/alexey/work/phpstan-src/tests/PHPStan/Command/AnalyseCommandTest.php:84
/home/alexey/work/phpstan-src/tests/PHPStan/Command/AnalyseCommandTest.php:25
/home/alexey/work/phpstan-src/vendor/phpunit/phpunit/src/Framework/TestCase.php:1154
```

The way to prevent this is to set the environment variable with a name constructed in `XdebugHandler::__construct` as
```php
$this->envAllowXdebug = self::$name.self::SUFFIX_ALLOW;
```
which results in the name of `PHPSTAN_ALLOW_XDEBUG`.

Reply:
Can you test this has the same effect? Thanks! https://github.com/phpstan/phpstan-src/commit/6fa7b67863d5f456de0ab1d8299510eca0e51a40

Reply:
Yes, it worked. You may want to add a comment there since as you said `PHPSTAN_ALLOW_XDEBUG` is not in the codebase.